### PR TITLE
Fix #7436: [hackerone] Sync settings should be behind Face ID / Passcode authentication

### LINKED ISSUE #7436: [hackerone] `Sync` settings should be behind Face ID / Passcode authentication
More details: https://hackerone.com/reports/1975155

When pressing `Sync` in settings users should be prompted with biometrics/passcode authentication.

### PR DESCRIPTION
**Pull request by soner-yuksel**
## Summary of Changes

This pull request fixes #7436

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

Test Plan 1: A user with no activated sync chain

Route 1: Settings Route

- Open Settings
- Click Sync
- Observe Sync Welcome page opens 
- All actions in the page should trigger Biometric Authentication

Route 2: Tab Tray Route

- Open TabTray and navigate Open tabs segmented part
- Click Start Sync Chain
- Observe Sync Welcome page opens 
- All actions in the page should trigger Biometric Authentication

Test Plan 2: A user with activated sync chain

Route 1: Settings Route

- Open Settings
- Click Sync
- Observe trigger Biometric Authentication
- After Authentication observer sync detail settings screen

Route 2: Tab Tray Route (an activated sync chain but open tabs not anabled)

- Open TabTray and navigate Open tabs segmented part
- Click Enable Tab Syncing
- Observe trigger Biometric Authentication
- After Authentication observer sync detail settings screen

## Screenshots:

https://github.com/brave/brave-ios/assets/6643505/508e111f-5d96-4f28-ac64-8bd33cd0e651

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveStrings/BraveStrings.swift [hunk 1]
```diff
@@ -3064,6 +3064,20 @@ extension Strings {
...
+        tableName: "BraveShared",
+        bundle: .module,
+        value: "Set a Passcode",
+        comment: "The title displayed in alert when a user needs to set passcode")
+    public static let syncSetPasscodeAlertDescription =
+      NSLocalizedString(
+        "login.loginInfoSetPasscodeAlertDescription",
+        tableName: "BraveShared",
+        bundle: .module,
+        value: "To setup sync hain or see settings, you must first set a passcode on your device..",
```
Comment (iccub) [inline Sources/BraveStrings/BraveStrings.swift]: `hain` typo

File: Sources/Brave/Frontend/Sync/SyncViewController.swift [hunk 2]
```diff
@@ -30,4 +41,41 @@ class SyncViewController: UIViewController {
...
+      }
+    }
+  }
+  
+  /// An alert presenter for passcode error to warn user to setup passcode to use feature
+  /// - Parameter completion: block after Ok button is pressed
+  private func showSetPasscodeError(completion: @escaping (() -> Void)) {
+    let alert = UIAlertController(
+      title: "Set a Passcode",
+      message: "To setup sync hain or see settings, you must first set a passcode on your device.",
```
Comment (iccub) [inline Sources/Brave/Frontend/Sync/SyncViewController.swift]: localize the alert

**Review by iccub — CHANGES_REQUESTED** [review]

File: Sources/BraveStrings/BraveStrings.swift (hunk 1 above)
Comment (soner-yuksel) [reply Sources/BraveStrings/BraveStrings.swift]: Fix at https://github.com/brave/brave-ios/pull/7450/commits/f6420db25399ef8c07a32d61bed45ef2182b960d

File: Sources/Brave/Frontend/Sync/SyncViewController.swift (hunk 2 above)
Comment (soner-yuksel) [reply Sources/Brave/Frontend/Sync/SyncViewController.swift]: Fix at https://github.com/brave/brave-ios/pull/7450/commits/f6420db25399ef8c07a32d61bed45ef2182b960d

**Review by iccub — APPROVED** [review]

File: Sources/Brave/Frontend/Sync/SyncViewController.swift [hunk 3]
```diff
@@ -30,4 +41,41 @@ class SyncViewController: UIViewController {
...
+      windowProtection.presentAuthenticationForViewController(
+        determineLockWithPasscode: false) { status in
+          completion?(status)
+      }
+    }
+  }
+  
+  /// An alert presenter for passcode error to warn user to setup passcode to use feature
+  /// - Parameter completion: block after Ok button is pressed
+  private func showSetPasscodeError(completion: @escaping (() -> Void)) {
```
Comment (soner-yuksel) [inline Sources/Brave/Frontend/Sync/SyncViewController.swift:67]: The biometrics doesnt exist error case alert.

**Pull request merged**